src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1150,11 +1150,27 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
 
         private string GetTablePrefix(string databaseName, string schemaName)
         {
-            StringBuilder tablePrefix = new(SqlQueryBuilder.QuoteIdentifier(databaseName));
-            if (!string.IsNullOrEmpty(schemaName))
+            StringBuilder tablePrefix = new();
+
+            if (!string.IsNullOrEmpty(databaseName))
```

Comment:
When both databaseName and schemaName are empty, this method will turn an empty string. The callers will append this empty string to 2 places, one of which: 
```csharp
string tablePrefix = GetTablePrefix(conn.Database, schemaName);
            selectCommand.CommandText
                = $"SELECT * FROM {tablePrefix}.{SqlQueryBuilder.QuoteIdentifier(tableName)}";
```

The references will need to be updated too.

Reply:
I guess for the purposes of this pr, database name will always be populated.

Reply:
This is a good point, what happens if both databaseName/schemaName are empty? 
the prefix will end up being empty, so either 
 1. the caller should check for emptiness of prefix before appending the `.` to {SqlQueryBuilder.QuoteIdentifier(tableName)} OR 
 2. this function should do the check and append the `.` to the tablePrefix before returning.

Reply:
Fixed above in latest iteration.